File: catchTheMousewithJava/app/src/main/java/com/abdbzkn/catchthemousewithjava/MainActivity.java
package com.abdbzkn.catchthemousewithjava;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.os.Handler;
import android.view.View;
import android.widget.ImageView;
import com.abdbzkn.catchthemousewithjava.databinding.ActivityMainBinding;
import java.util.Random;
public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    int score; int i; int j;
    ImageView[] imageArray; ImageView[] imageArray2;
    Runnable runnable; Handler handler;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        imageArray = new ImageView[] { binding.imageView, binding.imageView2, binding.imageView3, binding.imageView4, binding.imageView5, binding.imageView6, binding.imageView7, binding.imageView8, binding.imageView9, binding.imageView10, binding.imageView11, binding.imageView12, binding.imageView13, binding.imageView14, binding.imageView15};
        imageArray2 = new ImageView[] { binding.imageView16, binding.imageView17, binding.imageView18, binding.imageView19, binding.imageView20, binding.imageView21, binding.imageView22, binding.imageView23, binding.imageView24, binding.imageView25, binding.imageView26, binding.imageView27, binding.imageView28, binding.imageView29, binding.imageView30};
        score = 0;
        binding.resultText.setVisibility(View.INVISIBLE);
        new CountDownTimer(15300,1000){
            @Override
            public void onTick(long millisUntilFinished) {
                binding.timeText.setText("Time : " + millisUntilFinished/1000);
            }

            @Override
            public void onFinish() {
                binding.timeText.setText("Time is out");
                handler.removeCallbacks(runnable);
                for(ImageView image : imageArray){
                    image.setVisibility(View.INVISIBLE);
                }
                for(ImageView image : imageArray2){
                    image.setVisibility(View.INVISIBLE);
                }
                binding.scoreText.setVisibility(View.INVISIBLE);
                binding.resultText.setText("YOUR SCORE IS " + score + "!" );
                binding.resultText.setVisibility(View.VISIBLE);
            }
        }.start();
        hideImages();
    }
    public void takeScore(View view){
        j=i;
        score++;
        binding.scoreText.setText("Score : " + score);
        imageArray[j].setVisibility(View.INVISIBLE);
        new CountDownTimer(80,80){
            @Override
            public void onTick(long millisUntilFinished) {
              imageArray2[j].setVisibility(View.VISIBLE);
            }
            @Override
            public void onFinish() {
                imageArray2[j].setVisibility(View.INVISIBLE);
            }
        }.start();
    }
    public void hideImages(){
        handler = new Handler();
        runnable = new Runnable() {
            @Override
            public void run() {
                for(ImageView image : imageArray2){ //tüm imagelar görünmez yapıldı
                    image.setVisibility(View.INVISIBLE);
                }
                for(ImageView image : imageArray){
                    image.setVisibility(View.INVISIBLE);
                }
                Random random = new Random();
                i = random.nextInt(15);
                imageArray[i].setVisibility(View.VISIBLE);
                handler.postDelayed(this,550);
            }
        };
        handler.post(runnable);
        for(ImageView image : imageArray2){ //tüm imagelar görünmez yapıldı
            image.setVisibility(View.INVISIBLE);
        }
        for(ImageView image : imageArray){
            image.setVisibility(View.INVISIBLE);
        }
    }
}

File: catchTheMousewithKotlin/app/src/main/java/com/abdbzkn/catchthemousewithkotlin/MainActivity.kt
package com.abdbzkn.catchthemousewithkotlin

import android.content.DialogInterface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import android.os.Handler
import android.os.Looper
import android.view.View
import android.widget.ImageView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import com.abdbzkn.catchthemousewithkotlin.databinding.ActivityMainBinding
import java.util.Random

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    var score = 0
    val mouseimageArray = ArrayList<ImageView>()
    var runnable = Runnable{}
    var handler = Handler(Looper.getMainLooper())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        score = 0
        mouseimageArray.add(binding.imageView)
        mouseimageArray.add(binding.imageView2)
        mouseimageArray.add(binding.imageView3)
        mouseimageArray.add(binding.imageView4)
        mouseimageArray.add(binding.imageView5)
        mouseimageArray.add(binding.imageView6)
        mouseimageArray.add(binding.imageView7)
        mouseimageArray.add(binding.imageView8)
        mouseimageArray.add(binding.imageView9)
        mouseimageArray.add(binding.imageView10)
        mouseimageArray.add(binding.imageView11)
        mouseimageArray.add(binding.imageView12)
        hideImages()
        object : CountDownTimer(15300,1000){
            override fun onTick(millisUntilFinished: Long) {
                binding.timeText.text = "Time : ${millisUntilFinished/1000}"
            }
            override fun onFinish() {
                binding.timeText.text = "Time : 0"
                for(image in mouseimageArray){
                    image.visibility = View.INVISIBLE
                }
                handler.removeCallbacks(runnable)

                val alert = AlertDialog.Builder(this@MainActivity)
                alert.setTitle("Game Over")
                alert.setMessage("Restart The Game?")
                alert.setPositiveButton("Yes", DialogInterface.OnClickListener{ dialogInterface , i ->
                    //baştan başlatıcam
                    val intentFromMain = intent
                    finish()
                    startActivity(intentFromMain)
                })
                alert.setNegativeButton("No", DialogInterface.OnClickListener{ dialogInterface , i ->
                    //toast göstericmem
                    Toast.makeText(this@MainActivity,"Game Over",Toast.LENGTH_LONG).show()
                })
                alert.show()
            }
        }.start()
    }
    fun increaseScore(view : View){
        score++
        /* aynı konumda birden fazla score alınmaması için cok kısa bi süre engelledim
        object : CountDownTimer(90,10){
            override fun onTick(millisUntilFinished: Long) {
                for(image in mouseimageArray){
                    image.isEnabled = false
                }
            }
            override fun onFinish() {
                for(image in mouseimageArray){
                    image.isEnabled = true
                }
            }
        }.start()
        */
        binding.scoreText.text = "Score : ${score}"
    }
    fun hideImages(){
        runnable = object : Runnable{
            override fun run() {

                for ( image in mouseimageArray){
                    image.visibility = View.INVISIBLE
                }
                val random = Random()
                val randomIndex = random.nextInt(12)
                mouseimageArray[randomIndex].visibility = View.VISIBLE
                handler.postDelayed(runnable,400,)
            }
        }
        handler.post(runnable)
    }
}